### PR TITLE
Feature/complete extendible vs

### PR DESCRIPTION
**Pull request by huberya**

### CODE REVIEW

**Comment by HannesSommer** [comment]
There seems to be a consistent problem with the python exports:
```
/usr/include/boost/python/object/value_holder.hpp:137:13: error: no matching function for call to 'sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef()'
             BOOST_PP_REPEAT_1ST(N, BOOST_PYTHON_UNFORWARD_LOCAL, nil)
             ^

In file included from /home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_python/src/exportValueStoreRef.cpp:5:0:

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ValueStore::SharedPtr)
   using ValueStoreRef::ValueStoreRef;
                        ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ValueStore*)

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ValueStore&)

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:241:24: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:245:3: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::PropertyTree)
   ExtendibleValueStoreRef(sm::PropertyTree bpt);
   ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:245:3: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:244:12: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ExtendibleValueStore&)
   explicit ExtendibleValueStoreRef(ExtendibleValueStore & vs) : _evs(ExtendibleValueStore::SharedPtr(&vs, [](ExtendibleValueStore*){})) {}
            ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:244:12: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:243:12: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ExtendibleValueStore*)
   explicit ExtendibleValueStoreRef(ExtendibleValueStore * vsPtr) : _evs(ExtendibleValueStore::SharedPtr(vsPtr)) {}
            ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:243:12: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:242:3: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ExtendibleValueStore::SharedPtr)
   ExtendibleValueStoreRef(ExtendibleValueStore::SharedPtr spVs) : ValueStoreRef(spVs), _evs(spVs) {}
   ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:242:3: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:239:7: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(const sm::val...............................................................................
Failed << sm_python:make                    [ Exited with code 2 ]
ue_store::ExtendibleValueStoreRef&)
 class ExtendibleValueStoreRef : public ValueStoreRef {
       ^

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:239:7: note:   candidate expects 1 argument, 0 provided

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:239:7: note: candidate: sm::value_store::ExtendibleValueStoreRef::ExtendibleValueStoreRef(sm::value_store::ExtendibleValueStoreRef&&)

/home/jenkins/workspace/schweizer_messer/compiler/gcc/label/ubuntu-xenial/src/schweizer_messer/sm_value_store/include/sm/value_store/ValueStore.hpp:239:7: note:   candidate expects 1 argument, 0 provided
CMakeFiles/sm_python.dir/build.make:518: recipe for target 'CMakeFiles/sm_python.dir/src/exportValueStoreRef.cpp.o' failed


```

**Comment by huberya** [comment]
I can not reproduce the jenkins failure on my local machine...
However it looks to me as if it does not find the default constructor of extendible.
I tried to fix this by defining it explicitly instead of using Base::Base

File: sm_value_store/include/sm/value_store/ValueStore.hpp [hunk 1]
```diff
@@ -240,24 +238,26 @@ class ExtendibleValueStore : public ValueStore {
 
 class ExtendibleValueStoreRef : public ValueStoreRef {
  public:
-  ExtendibleValueStoreRef(sm::PropertyTree bpt);
+  ExtendibleValueStoreRef() : ValueStoreRef() {}
```
Comment (HannesSommer) [inline sm_value_store/include/sm/value_store/ValueStore.hpp]: Why no longer accept a property tree?

**Comment by HannesSommer** [comment]
Oops, I merged a bit early. If you revive the `ExtendibleValueStoreRef` constructor taking a PT, we can merge that in a next step.

File: sm_value_store/include/sm/value_store/ValueStore.hpp (hunk 1 above)
Comment (huberya) [reply sm_value_store/include/sm/value_store/ValueStore.hpp]: It just moved a bit down, in consistency with ValueStoreRef (line 245)

Comment (HannesSommer) [reply sm_value_store/include/sm/value_store/ValueStore.hpp]: Ah, great! Then my merging wasn't prematurely after all :)!

**Pull request merged**